mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,328 @@
+ * @include OpenLayers/Format/Filter.js
+ * @include OpenLayers/Format/XML.js
+ // see https://github.com/georchestra/georchestra/issues/482 :
+ * @include OpenLayers/Format/WKT.js
+ * @include OpenLayers/Control/ModifyFeature.js
+ * @include OpenLayers/Control/DrawFeature.js
+ * @include OpenLayers/Handler/Point.js
+ * @include OpenLayers/Handler/Path.js
+ * @include OpenLayers/Handler/Polygon.js
+ * @include GeoExt/data/LayerRecord.js
+ * @include GeoExt/data/AttributeStore.js
+ * @include Styler/widgets/FilterBuilder.js
+ * @include GEOR_waiter.js
+ * @include GEOR_ows.js
+ * @include GEOR_util.js
+ */
+
+/* globals Ext, GeoExt, OpenLayers, GEOR, Styler */
+
+Ext.namespace("GEOR");
+
+
+GEOR.Querier = Ext.extend(Ext.Window, {
+
+    layout: "fit",
+    closeAction: "close",
+    border: false,
+ 
+    /**
+     * Property: filterbuilder
+     * {String} The local Styler.FilterBuilder instance
+     */
+    filterbuilder: null,
+
+    /**
+     * Property: record
+     * {Ext.data.Record} The WMS or WFS layer record
+     */
+    record: null,
+
+    /**
+     * Property: map
+     * {OpenLayers.Map} The map instance.
+     */
+    map: null,
+
+    /**
+     * Property: layer
+     * {OpenLayers.Layer.Vector} The layer on which geometries can be drawn
+     */
+    layer: null,
+
+    /**
+     * Property: attributeStore
+     * {GeoExt.data.AttributeStore} 
+     */
+    attributeStore: null,
+ 
+    /**
+     * Property: _geometryName
+     * {String} The layer geometry name, obtained by DescribeFeatureType
+     */
+    _geometryName: null,
+
+    /*
+     * Method: initComponent.
+     * Overridden constructor. Set up widgets and lay them out
+     */
+    initComponent: function() {
+        var r = this.record,
+            type = r.get("type"),
+            isWFS = type === "WFS",
+            layer = r.get("layer"), 
+            name = r.get("title") || layer.name || "";
+
+        this.title = OpenLayers.i18n("Request on NAME", {
+            "NAME": name
+        });
+
+        var pseudoRecord = {
+            typeName: isWFS ? 
+                r.get("WFS_typeName") : r.get("name"),
+            owsURL: isWFS ? 
+                layer.protocol.url : r.get("WFS_URL")
+        };
+
+        GEOR.waiter.show();
+        // get layer model through WFS DescribeFeatureType:
+        this.attributeStore = GEOR.ows.WFSDescribeFeatureType(pseudoRecord, {
+            extractFeatureNS: true,
+            success: function() {
+                // we get the geometry column name, and remove the corresponding record from store
+                var idx = this.attributeStore.find("type", GEOR.ows.matchGeomProperty);
+                if (idx > -1) {
+                    // we have a geometry
+                    var r = this.attributeStore.getAt(idx),
+                        geometryName = r.get("name");
+                    // create the protocol:
+                    this.protocol = GEOR.ows.WFSProtocol(pseudoRecord, this.map, {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://www.codacy.com/app/georchestra/georchestra/file/2338524863/issues/source?bid=3200181&fileBranchId=3209168#l130)